src/components/CreateNewBudgetSheet.js

```diff
@@ -0,0 +1,177 @@
+import { useState } from 'react';
+import firebase from '../firebase';
+import { getDatabase, ref, push } from 'firebase/database';
+import { useNavigate } from 'react-router-dom';
+import WarningModal from './WarningModal';
+
+const NewBudgetSheet = (props) => {
+  const database = getDatabase(firebase);
+  const dbRef = ref(database);
+
+  const [name, setName] = useState('');
```

Comment:
Let me preface this by saying I haven't look at the rest of the app yet. I wonder whether it would be useful or worthwhile to use a single state for all of these, where the state holds a single 'budget' object that has these as its properties. It might not make the app more readable or maintainable, so maybe not.

Reply:
that's a good idea. I haven't looked into redux yet, but maybe that's another solution to managing all these states i'm creating lol